scripts/py_matter_yamltests/matter_yamltests/errors.py

```diff
@@ -0,0 +1,163 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the 'License');
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an 'AS IS' BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import yaml
+
+_ERROR_START_TAG = '__error_start__'
+_ERROR_END_TAG = '__error_end__'
+
+
+class TestStepError(Exception):
+    """Raise when a step is malformed."""
+
+    def __init__(self, message):
+        self.step_index = 0
+        self.context = None
+        self.message = message
+
+    def __str__(self):
+        return self.message
+
+    def update_context(self, context, step_index):
+        self.context = yaml.dump(
+            context,
+            default_flow_style=False,
+            sort_keys=False
+        )
+        self.step_index = step_index
+
+    def tag_key_with_error(self, content, target_key):
+        self.__tag_key(content, target_key, _ERROR_START_TAG, _ERROR_END_TAG)
+
+    def __tag_key(self, content, target_key, tag_start, tag_end):
+        reversed_dictionnary = {}
+
+        # Build a reversed dictionnary, tagging the target key.
```

Comment:
So we're guaranteed this is using Python 3.7 or newer?

Reply:
I made it like that because pigweed is taking `Python 3.9.5+chromium.19` so will double check with our test harness friends which version of python they are using and open a followup if needed.

Reply:
The dockers image running on the RPI seems to tun `Python 3.10.6` so that sounds good.